x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/textstructure/action/FindStructureAction.java

```diff
@@ -262,6 +274,16 @@ public void setGrokPattern(String grokPattern) {
             this.grokPattern = (grokPattern == null || grokPattern.isEmpty()) ? null : grokPattern;
         }
 
+        public String getEcsCompatibility() {
+            return ecsCompatibility;
+        }
+
+        public void setEcsCompatibility(String ecsCompatibility) {
+            this.ecsCompatibility = (ecsCompatibility == null || ecsCompatibility.isEmpty())
+                ? ECS_COMPATIBILITY_DISABLED
+                : ecsCompatibility;
+        }
```

Comment:
This is inconsistent with other code that sets `ecsCompatibility`.  The other code, for example the parser and both constructors, allow it to be `null`.  This method converts `null` to `"disabled"`.  It should be handled consistently so that either everywhere where `ecsCompatibility` is set uses `disabled` instead of `null` or else this setting mechanism considers `null` a separate value.